### DISCUSSION TITLE
second pass aka hires fix via API?

### DISCUSSION THREAD

**Discussion by blacklig** [vladmandic/sdnext#2262] [Q&A]
Hi, can anyone help me how to activate second pass aka high res fix via api?

I am trying for example those:

Payload = {
                "prompt": imageData.prompt,
                "negative_prompt": "fewer digits, cropped, worst quality, low quality, unrealistic size, bad dimensions, bad perspective",
                "seed": -1,
                "subseed": -1,
                "subseed_strength": 0,
                "sampler_name": "DDIM",
                "batch_size": 1,
                "n_iter": 1,
                "cfg_scale": 6,

                "width": 512,
                "height": 384,
                "steps": 10,

                "hr_scale": 2,
                "hr_force": true,
                "hr_upscaler": "R-ESRGAN 2x+",
                "hr_second_pass_steps": 20,
                "full_quality": true,
                "hr_resize_x": 1024,
                "hr_resize_y": 768,
            }

I still can not get it how to activate that R-ESRGAN 2x+ ... or anyone can give me idea how to "reverse engineer" what next SD sends? Unfortunatelly it is not visible on console even if --debug is turned on.. Tried to find it in code of front end but no success... I am bit lost here :) Help appreciated

**Comment by vladmandic** [comment]
there's a feature request to provide clean api for second pass, i should add it soon.

**Reply by blacklig** [reply]
and is there for now some dirty way at least? :)

I am trying to somehow reverse engineer it from front end, but most I get is something like this:

txt2img_paste_fields = [
                (txt2img_prompt, "Prompt"),
                (txt2img_negative_prompt, "Negative prompt"),
                (steps, "Steps"),
                (sampler_index, "Sampler"),
                (restore_faces, "Face restoration"),
                (cfg_scale, "CFG scale"),
                (seed, "Seed"),
                (width, "Size-1"),
                (height, "Size-2"),
                (batch_size, "Batch size"),
                (subseed, "Variation seed"),
                (subseed_strength, "Variation seed strength"),
                (seed_resize_from_w, "Seed resize from-1"),
                (seed_resize_from_h, "Seed resize from-2"),
                (denoising_strength, "Denoising strength"),
                (enable_hr, lambda d: "Denoising strength" in d),
                (hr_options, lambda d: gr.Row.update(visible="Denoising strength" in d)),
                (hr_scale, "Hires upscale"),
                (hr_upscaler, "Hires upscaler"),
                (hr_second_pass_steps, "Hires steps"),
                (hr_resize_x, "Hires resize-1"),
                (hr_resize_y, "Hires resize-2"),
                *modules.scripts.scripts_txt2img.infotext_fields
            ]
            
            in modules/ui.py

but even "txt2img_prompt" is not compatible with api somehow, as api wants just "prompt".. So I suppose even some other variables might be different than this :/

**Comment by vladmandic** [comment]
for api, don't try to second-guess or reverse-engineer code, just start `webui --docs` and look at generated docs.
and based on those, it took me ~1min to come up with an end-to-end working example:
<https://github.com/vladmandic/automatic/blob/dev/cli/simple-txt2img.js>
